test/slotted.html

```diff
@@ -0,0 +1,195 @@
+          <paper-input name="paper2" value=""></paper-input>
+          <input type="reset">
+        </div>
+      </slotted-form>
+    </template>
+  </test-fixture>
+
+
+
+  <script>
+    suite('slotted', function() {
+      var form;
+      var server;
+
+      setup(function() {
+        server = sinon.fakeServer.create();
+        server.respondWith(
+          'GET',
+          /\/get.*/,
+          [
+            200,
+            '{"Content-Type":"application/json"}',
+            '{"success":true}'
+          ]
+        );
+      });
+
+      teardown(function() {
+        server.restore();
+      });
+
+      suite('serialization', function() {
+        setup(function() {
+          form = fixture('slotted').$.form;
+        });
+        test('serializes both normal and distributed nodes', function(done) {
+          form.addEventListener('iron-form-response', function(event) {
+            var expectedUrl = encodeURI('/get?original=works&checkbox=on&input=some');
+            expect(event.detail.url).to.equal(expectedUrl);
+            expect(event.detail.response.success).to.be.equal(true);
+            done();
+          });
+
+          // Wait one tick for observeNodes.
+          Polymer.Base.async(function() {
+            form.submit();
+            server.respond();
+          }, 1);
+        });
+      });
+
+      suite('validation', function() {
+        setup(function() {
+          form = fixture('slotted-validation').$.form;
+        });
+        test('<paper-checkbox required> is validated and does not submit the form', function(done) {
+          var responses = 0;
+          form.addEventListener('iron-form-response', function(event) {
+            responses++;
+          });
+
+          // Wait one tick for observeNodes.
+          Polymer.Base.async(function() {
+            expect(form.validate()).to.be.equal(false);
+            form.submit();
+            server.respond();
+          }, 1);
+
+          setTimeout(function() {
+            expect(responses).to.be.equal(0);
+            done();
+          },  200);
+        });
+      });
+    });
+
+    suite('resetting', function() {
+      var inputs;
+      setup(function() {
+        form = fixture('slotted-reset').$.form;
+        var slot = form.querySelector('slot,content');
+        if (slot) {
+          inputs = Polymer.dom(slot).getDistributedNodes()[0];
+        } else {
+          inputs = form.querySelector('[slot]');
+        }
+      });
+
+      test('can reset a form', function(done) {
+        // Wait one tick for observeNodes.
+        Polymer.Base.async(function() {
+          var initial = form.serializeForm();
+          expect(JSON.stringify(initial)).to.be.equal('{"original":"works","input1":"input1","input2":"","check1":"on","radio1":"on","papercheck1":"on","paper1":"paper1","paper2":""}');
+
+          // Modify all the values, flip all the inputs.
+          form.querySelector('[name=original]').value = 'works++';
```

Comment:
The following lines are missing trailing `]` characters in the selectors.  This makes Safari mad at least.
```
index 4356c36..f407148 100644
--- a/test/slotted.html
+++ b/test/slotted.html
@@ -168,16 +168,16 @@ subject to an additional IP rights grant found at http://polymer.github.io/PATEN

           // Modify all the values, flip all the inputs.
           form.querySelector('[name=original]').value = 'works++';
-          inputs.querySelector('[name=input1').value = 'input1++';
-          inputs.querySelector('[name=input2').value = 'input2++';
-          inputs.querySelector('[name=check1').checked = false;
-          inputs.querySelector('[name=check2').checked = true;
-          inputs.querySelector('[name=radio1').checked = false;
-          inputs.querySelector('[name=radio2').checked = true;
-          inputs.querySelector('[name=papercheck1').checked = false;
-          inputs.querySelector('[name=papercheck2').checked = true;
-          inputs.querySelector('[name=paper1').value = 'paper1++';
-          inputs.querySelector('[name=paper2').value = 'paper2++';
+          inputs.querySelector('[name=input1]').value = 'input1++';
+          inputs.querySelector('[name=input2]').value = 'input2++';
+          inputs.querySelector('[name=check1]').checked = false;
+          inputs.querySelector('[name=check2]').checked = true;
+          inputs.querySelector('[name=radio1]').checked = false;
+          inputs.querySelector('[name=radio2]').checked = true;
+          inputs.querySelector('[name=papercheck1]').checked = false;
+          inputs.querySelector('[name=papercheck2]').checked = true;
+          inputs.querySelector('[name=paper1]').value = 'paper1++';
+          inputs.querySelector('[name=paper2]').value = 'paper2++';

           var updated = form.serializeForm();
           expect(JSON.stringify(updated)).to.not.be.equal(initial);
```

Reply:
Pushed 1145193c2924e766f41224f46137e731aa2a24d8 to fix.

Reply:
@notwaldorf Tests fixed.

Reply:
Oh awesome, totally missed that. Thanks for fixing!

Reply:
I have filed https://github.com/whatwg/dom/issues/549 to address the inconsistency between browsers.